### DISCUSSION TITLE
M3, M4 buttons sometimes don't work

### DISCUSSION THREAD

**Discussion by thduc87** [seerge/g-helper#803] [Q&A]
Hi, i just recently encountered this issue, boot up laptop, the first few seconds i can still use M3, M4 buttons but then they are completely disabled and i have to restart g-helper, any idea why? i'm on 0.95

**Comment by seerge** [comment]
@thduc87 post a log from ``%appdata%\ghelper``and what things from asus do you have running at least . I don't have extrasensory skills :)

**Reply by thduc87** [reply]
hi sorry for late reply, i have updated to latest stable release 0.100 but still occasionally experience this issue, i have attached a log file here for your review, thanks.
[log.txt](https://github.com/seerge/g-helper/files/12144273/log.txt)

**Reply by thduc87** [reply]
the problem occured again around 4.22pm until end of log.

**Reply by thduc87** [reply]
i disabled all asus services in ghelper.

**Reply by seerge** [reply]
@thduc87 hello, i see that your keyboard (actually windows) by some reason has terminated connection with the app by some reason

```
7/24/2023 4:22:19 PM: Listener terminated
```

This is normal to happen when laptop goes to sleep (you would see lots of entries like that in your log after "monitor off" entries, but then when laptop wakes up app always re-establishes connection with keyboard. 

I don't know why it happened at 4:22 tho :) may be windows having own issues? Do you have any tricky sleep settings? Try to select default windows power plan (i think it's balanced in control panel) and reset it to defaults as well.

**Reply by thduc87** [reply]
i only have the default balanced mode, i did tweak the avanced setting to set the maximum processor state to 80%., i'll reset to default and see if the problem still persists, thanks.

**Comment by kielhanson** [comment]
I've just bought a (2nd hand) ROG Zephyrus G14 (2022). The M keys haven't worked at all since I got it. I just discovered G Helper thinking it may resolve this issue but it seems they also don't work when using G Helper and disabling Armoury Crate :(

**Reply by seerge** [reply]
@kielhanson looks like they are broken on hardware level in your case then

**Reply by seerge** [reply]
Did you stop asus services from Extra -> Stop?

**Comment by kielhanson** [comment]
> Did you stop asus services from Extra -> Stop?

Yes, ASUS Services running: 0 is what it says. I've removed Armoury Crate and MyAsus entirely, still no joy :( 
Seems they are completely broken ...

**Comment by kohanyirobert** [comment]
I think I have a similar problem.

I'm currently on battery,
Noticed that M4 doesn't work.
Found this post.
Restarted G-Helper, started tailing the logs (attached).
Still M4 doesn't work, my keyboard backlight doesn't start (FN+Backlight buttons don't respond),

I think it's related, but sometimes the laptop can get in a state where some keypresses only register the second time I press them. Primarily Alt+Tab, or at least that's the most noticable. It's not a hardware issue, I don't have to press it harder the second time. (Just as I'm writing this Ctrl+T didn't work first when I wanted to open another tab.)

No idea what can cause it. No sticky keys enabled, drivers up to date. My ASUS services are stopped.

```
2024. 09. 27. 15:41:09: App launched: ROG Zephyrus G14 GA402RK_GA402RK :0.191.0.0en-US
2024. 09. 27. 15:41:09: Start Count: 52
2024. 09. 27. 15:41:09: Matrix Device: \\?\hid#vid_0b05&pid_193b&col02#7&2ce29178&0&0001#{4d1e55b2-f16f-11cf-88cb-001111000030} 640
2024. 09. 27. 15:41:10: dGPU not found
2024. 09. 27. 15:41:12: CPU: 8 - AMD Ryzen 7 6800HS with Radeon Graphics          - AMD64 Family 25 Model 68 Stepping 1
2024. 09. 27. 15:41:12: AutoSetting for Offline
2024. 09. 27. 15:41:12: BatteryLimit = 60 : OK
2024. 09. 27. 15:41:12: Input available: \\?\hid#vid_0b05&pid_19b6&mi_00&col01#8&18c99fbc&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030} 19B6 64
2024. 09. 27. 15:41:12: USB 19B6|64: 5A-41-53-55-53-20-54-65-63-68-2E-49-6E-63-2E
2024. 09. 27. 15:41:12: Input: \\?\hid#vid_0b05&pid_19b6&mi_00&col01#8&18c99fbc&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}
2024. 09. 27. 15:41:12: Eco flag : 1
2024. 09. 27. 15:41:12: Mux flag : 1
2024. 09. 27. 15:41:12: \\?\DISPLAY#TMX1481#5&2e73fae4&1&UID1024#{e6f07b5f-ee97-4a90-b076-33f57bf4eaa7} DISPLAYCONFIG_OUTPUT_TECHNOLOGY_INTERNAL
2024. 09. 27. 15:41:12: \\?\DISPLAY#TMX1481#5&2e73fae4&1&UID1024#{e6f07b5f-ee97-4a90-b076-33f57bf4eaa7} \\.\DISPLAY1\Monitor0
2024. 09. 27. 15:41:12: TL140ADXP01 HDR: False 8 13 True
2024. 09. 27. 15:41:12: Mode = 2 : OK
2024. 09. 27. 15:41:12: Matrix Device: \\?\hid#vid_0b05&pid_193b&col02#7&2ce29178&0&0001#{4d1e55b2-f16f-11cf-88cb-001111000030} 640
2024. 09. 27. 15:41:12: Init 19B6: 5D-B9
2024. 09. 27. 15:41:12: Init 19B6: 5D-41-53-55-53-20-54-65-63-68-2E-49-6E-63-2E
2024. 09. 27. 15:41:12: Init 19B6: 5D-05-20-31-00-1A
2024. 09. 27. 15:41:12: USB 19B6: 5D-BD-01-5D-1F-FF-FF-FF
2024. 09. 27. 15:41:12: AuraMode: AuraStatic
2024. 09. 27. 15:41:12: USB 19B6: 5D-B3-00-00-FF-FF-FF-EB-00-00-FF-FF-FF-00-00-00-00
2024. 09. 27. 15:41:12: USB 19B6: 5D-B5-00-00-00
2024. 09. 27. 15:41:12: USB 19B6: 5D-B4
2024. 09. 27. 15:41:12: Matrix Off
2024. 09. 27. 15:41:12: Monitor Power On
2024. 09. 27. 15:41:12: BatteryLimit = 60 : OK
2024. 09. 27. 15:41:12: Lid Open
2024. 09. 27. 15:41:12: Lid 19B6: 5D-BA-C5-C4-01
2024. 09. 27. 15:41:13: Auto 19B6: 5D-BA-C5-C4-01
2024. 09. 27. 15:41:14: Timeout 19B6: 5D-BA-C5-C4-00
2024. 09. 27. 15:41:15: Auto 19B6: 5D-BA-C5-C4-01
2024. 09. 27. 15:41:22: Error writing Timeout 19B6: Operation failed early: A device which does not exist was specified. 5D-BA-C5-C4-00
2024. 09. 27. 15:41:22: Listener exited: Operation failed after some time.
2024. 09. 27. 15:41:23: Auto 19B6: 5D-BA-C5-C4-01
2024. 09. 27. 15:41:23: HID Device Event: Checking for new ASUS Mice
2024. 09. 27. 15:41:28: Timeout 19B6: 5D-BA-C5-C4-00
2024. 09. 27. 15:41:31: Auto 19B6: 5D-BA-C5-C4-01
2024. 09. 27. 15:41:41: HID Device Event: Checking for new ASUS Mice
2024. 09. 27. 15:42:13: HID Device Event: Checking for new ASUS Mice
2024. 09. 27. 15:42:44: HID Device Event: Checking for new ASUS Mice
2024. 09. 27. 15:42:55: HID Device Event: Checking for new ASUS Mice
2024. 09. 27. 15:42:59: TL140ADXP01 HDR: False 8 13 True
2024. 09. 27. 15:43:01: Latest version 0.191.0.0
2024. 09. 27. 15:43:05: Varibright: 1,2
2024. 09. 27. 15:43:05: Hibernate after 2147483647
2024. 09. 27. 15:43:46: HID Device Event: Checking for new ASUS Mice
2024. 09. 27. 15:44:30: Error writing Timeout 19B6: Operation failed early: A device which does not exist was specified. 5D-BA-C5-C4-00
2024. 09. 27. 15:44:31: HID Device Event: Checking for new ASUS Mice
2024. 09. 27. 15:44:35: Auto 19B6: 5D-BA-C5-C4-01
2024. 09. 27. 15:45:19: HID Device Event: Checking for new ASUS Mice
2024. 09. 27. 15:46:08: Timeout 19B6: 5D-BA-C5-C4-00
2024. 09. 27. 15:46:26: Error writing Auto 19B6: Operation failed early: A device which does not exist was specified. 5D-BA-C5-C4-01
2024. 09. 27. 15:46:27: HID Device Event: Checking for new ASUS Mice
2024. 09. 27. 15:46:35: HID Device Event: Checking for new ASUS Mice
```

**Reply by seerge** [reply]
@kohanyirobert looks like keyboard is misbehaving on hardware level as USB device. It simply keeps disconnecting / connecting all the time (in windows). As this events `2024. 09. 27. 15:41:41: HID Device Event` - are just sent by windows, every time new USB device connects or disconnects.

And this means that app is not able to reach keyboard to set backlight settings and can't listen for incoming macro-hotkeys (like M4)
```
2024. 09. 27. 15:41:22: Error writing Timeout 19B6: Operation failed early: A device which does not exist was specified. 5D-BA-C5-C4-00
2024. 09. 27. 15:41:22: Listener exited: Operation failed after some time.
```

Try to do [hard reset to see if it helps. ](https://github.com/seerge/g-helper/wiki/Troubleshooting#hard-reset--embedded-controller-ec-reset)

**Reply by kohanyirobert** [reply]
I did a regular restart (before your comment) and it seems to work, no `HID Device Event: Checking for new ASUS Mice` or `Error writing Auto 19B6`. Will try hard reset if this becomes a mainstay :S

When it happens again tho', any chance I could figure out which device is the one exactly that causes the problem?
I don't quite get what to look for based on G-Helper's logs (not sure what the strings looking like device IDs can be matched with device attributes I can find in Windows). If you have any tips let me know. Maybe there's something to do here if somebody can figure this out.

**Reply by seerge** [reply]
@kohanyirobert i think it's keyboard itself that is disconnecting (it's just another "usb-device" in windows, even tho it's internal). But since it works fine after restart - then good :)
